sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/ObjectSerializer.java

```diff
@@ -22,28 +40,69 @@
      */
     <T> T deserialize(InputStream stream, TypeReference<T> typeReference);
 
+    /**
+     * Reads a byte array into its object representation.
+     *
+     * @param data Byte array.
+     * @param typeReference {@link TypeReference} representing the object.
+     * @param <T> Type of the object.
+     * @return Reactive stream that emits the object represented by the deserialized byte array.
+     */
+    default <T> Mono<T> deserializeAsync(byte[] data, TypeReference<T> typeReference) {
```

Comment:
`typeReference` : Should we check for null and return Mono.error ?

Reply:
Given that the original interface leaves that up for interpretation I'd leave it to the implementations to decide.